libraries/AP_Follow/AP_Follow.cpp

```diff
@@ -611,19 +624,48 @@ bool AP_Follow::handle_global_position_int_message(const mavlink_message_t &msg)
         return false;
     }
 
+    // apply jitter-corrected timestamp to this update
+    uint32_t location_update_ms = _jitter.correct_offboard_timestamp_msec(packet.time_boot_ms, AP_HAL::millis());
+    if (location_update_ms < _last_location_update_ms) {
+        // ignore if the new update is older than the most recent one received
+        return false;
+    }
+    _last_location_update_ms = location_update_ms;
+
     Location _target_location;
     _target_location.lat = packet.lat;
     _target_location.lng = packet.lon;
 
-    // set target altitude based on configured altitude type
-    if (_alt_type == AP_FOLLOW_ALTITUDE_TYPE_RELATIVE) {
-        // use relative altitude above home
+#if APM_BUILD_TYPE(APM_BUILD_ArduPlane|APM_BUILD_ArduCopter)
```

Comment:
I think you could just if def out the terrain case in the switch. The abs and home cases are the same I think?

Reply:
Agreed, makes sense.